WBooks/Library/MainMenuViewModel.swift

```diff
@@ -12,20 +12,24 @@ class MainMenuViewModel {
     
     var books: [BookViewModel] = []
     
+    var menuController: MainMenuController?
+    
     init() {
-        books = createArray()
+        BookRepository.fetchBooks(onSuccess: onSuccess, onError: onError)
```

Comment:
You must call this when the View Controller asks for the books, not each time the view model is created.

Reply:
It would be nice to have a repository property, and use it, not a static function. This way, when we need to add test to our app, it is as simple as creating a different repository and use it.